source/simulator/solver.cc

```diff
@@ -861,7 +869,8 @@ namespace aspect
 
     // do some cleanup now that we have the solution
     remove_nullspace(solution, distributed_stokes_solution);
-    this->last_pressure_normalization_adjustment = normalize_pressure(solution);
+    if (parameters.nonlinear_solver != NonlinearSolver::Newton_Stokes)
+      this->last_pressure_normalization_adjustment = normalize_pressure(solution);
```

Comment:
but do you do the normalization at the end of *all* Newton iterations? this call must be somewhere!